src/github/pullRequestGitHelper.ts

```diff
@@ -54,40 +54,45 @@ export class PullRequestGitHelper {
 
 	static async fetchAndCheckout(repository: Repository, remote: Remote, branchName: string, pullRequest: IPullRequestModel): Promise<void> {
 		let remoteName = remote.remoteName;
-		Logger.debug(`Fetch remote ${remoteName}`, PullRequestGitHelper.ID);
-		await repository.fetch(remoteName);
-
 		let branch: Branch;
 
 		try {
 			branch = await repository.getBranch(branchName);
```

Comment:
This looks good, but I wonder if we can simplify the logic more. The `fetchAndCheckout` and `createAndCheckout` methods repeat a lot of the same operations and when I was first reading them I was confused about when each should be called, based on the name I wasn't sure what would be created in `createAndCheckout`.

Reply:
Thanks for your suggestion. We should be able to do that. The way we switch branches right now

* Validate local branch to see if there is already one, and the return value is simply branch name and remote name, which are strings.
  * Fetch the branch and checkout if it exits
  * Create remtoe and then fetch

It works fine if Git is no cost but unluckily that's not true. Instead of having a `if` condition check, which is a costly git operation, we should just fetch the whole config, find all the information needed, if we find something matched, checkout; if not, go to `createAndCheckout`.